src/cljs/org/broadinstitute/firecloud_ui/page/workspaces_list.cljs

```diff
@@ -16,47 +16,60 @@
-         [input/TextField {:ref "wsNamespace" :style {:width "100%"}
-                           :defaultValue "broad-dsde-dev"
-                           :predicates [(input/nonempty "Google Project")]}]
-         (style/create-form-label "Name")
-         [input/TextField {:ref "wsName" :style {:width "100%"}
-                           :predicates [(input/nonempty "Workspace name")]}]
-         (style/create-form-label "Description (optional)")
-         (style/create-text-area {:style {:width "100%"} :rows 5 :ref "wsDescription"})
-         [comps/ErrorViewer {:error (:server-error @state)}]
-         (style/create-validation-error-message (:validation-error @state))])
-      :dismiss-self #(swap! state dissoc :overlay-shown? :server-error :validation-error)
-      :ok-button
-      (react/create-element
-        [comps/Button
-         {:text "Create Workspace" :ref "createButton"
-          :onClick #(let [[ns n & fails] (input/get-and-validate refs "wsNamespace" "wsName")]
-                      (if fails
-                        (swap! state assoc :validation-error fails :server-error nil)
-                        (let [desc (common/get-text refs "wsDescription")
-                              attributes (if (clojure.string/blank? desc) {} {:description desc})]
-                          (swap! state assoc :creating-wf true :server-error nil :validation-error nil)
-                          (endpoints/call-ajax-orch
-                            {:endpoint (endpoints/create-workspace ns n)
-                             :payload {:namespace ns :name n :attributes attributes}
-                             :headers {"Content-Type" "application/json"}
-                             :on-done (fn [{:keys [success? get-parsed-response]}]
-                                        (swap! state dissoc :creating-wf)
-                                        (if success?
-                                          (do (swap! state dissoc :overlay-shown? :server-error :validation-error)
-                                            (nav/navigate nav-context (str ns ":" n)))
-                                          (swap! state assoc :server-error (get-parsed-response))))}))))}])}]))
+(react/defc CreateWorkspaceDialog
+  {:get-initial-state
+   (fn [{:keys [props]}]
+     {:selected-project (first (:billing-projects props))})
+   :render
+   (fn [{:keys [props state refs this]}]
+     [dialog/Dialog
+      {:width 500
+       :dismiss-self (:dismiss props)
+       :content
+       (react/create-element
+         [dialog/OKCancelForm
+          {:header "Create New Workspace"
+           :content
+           (react/create-element
+             [:div {:style {:marginBottom -20}}
+              (when (:creating-wf @state)
+                [comps/Blocker {:banner "Creating Workspace..."}])
+              (style/create-form-label "Google Project")
+              (style/create-select {:onChange #(swap! state assoc :selected-project (-> % .-target .-value))}
```

Comment:
I think you should also include `:value (:selected-project @state)` here. Otherwise, if we ever change `:get-initial-state` to be something besides the first one, this will break.